modules/huds/hudsAccesos.ts

```diff
@@ -0,0 +1,70 @@
+import { HudsAcceso } from './hudsAccesos.schema';
+import * as moment from 'moment';
+
+
+export async function logAcceso(req, paciente, matricula, motivo, idTurno, idPrestacion) {
+    let bucketNumber = 0;
+    let retry = true;
+    while (retry) {
+        try {
+            await execLog(req, paciente, matricula, motivo, idTurno, idPrestacion, bucketNumber);
+            retry = false;
+        } catch (err) {
+            if (err.code === 17419) {
+                bucketNumber++;
+            } else {
+                retry = false;
+                throw new Error(err);
+            }
+        }
+    }
+}
+
+async function execLog(req, paciente, matricula, motivoAcceso, turno, prestacion, bucketNumber) {
+    const now = new Date();
+    return HudsAcceso.updateOne(
+        {
+            paciente,
+            anio: moment(now).year(),
```

Comment:
usuario un parametro más generico por si se decide cambiar el tamaño de backet

```suggestion
            start: moment(now).startOf('year'),
```

Reply:
> Esta porción de código estaría bueno que sea un middleware.
> 
> ```
> let tokenSettings;
>     if (req.query.hudsToken) {
>         tokenSettings = Auth.decode(req.query.hudsToken);
>     }
> ```
> 
> Esta muy repetido por toda la app.

Se agrega middlware en commit https://github.com/andes/api/pull/843/commits/379fd4fed3816cb4160b5d87a844cd927d6ed434